pkg/ingester/ingester.go

```diff
@@ -1386,6 +1417,14 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 	var builder labels.ScratchBuilder
 	var nonCopiedLabels labels.Labels
 	for _, ts := range timeseries {
+		// The cost attribution value for the series
+		costLabel := i.limits.CostAttributionLabel(userID)
```

Comment:
There's no need to call this for each timeseries
```suggestion
	// The cost attribution value for the series
	costLabel := i.limits.CostAttributionLabel(userID)
	for _, ts := range timeseries {
```